core/actionProxy/actionproxy.py

```diff
@@ -136,15 +136,19 @@ def error(msg):
         # run the process and wait until it completes.
         (o, e) = p.communicate()
 
-        if o is not None:
-            process_output_lines = o.strip().split('\n')
-            last_line = process_output_lines[-1]
-            for line in process_output_lines[:-1]:
-                sys.stdout.write('%s\n' % line)
-        else:
-            last_line = '{}'
-
-        if e is not None:
+        # stdout/stderr may be either text or bytes, depending on Python
+        # version.   In the latter case, decode to text.
+        if isinstance(o, bytes):
+            o = o.decode('utf-8')
+        if isinstance(e, bytes):
+            e = e.decode('utf-8')
+
+        process_output_lines = o.strip().split('\n')
```

Comment:
I am sorry.  Perhaps my writing was unclear.  I think we still need the if/then/else.  The change that I propose was just to change `if o is not None:` --> `if o:`.